fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -805,214 +843,377 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
+	int latencyBucket = 0;
 
-	// Active load balancing runs at a very high priority (to obtain accurate estimate of memory used by commit batches) so we need to downgrade here
-	wait(delay(0, TaskPriority::ProxyCommit));
+	Version commitVersion;
+	Version prevVersion;
+
+	int64_t maxTransactionBytes;
+	std::vector<std::vector<int>> transactionResolverMap;
+	std::vector<std::vector<std::vector<int>>> txReadConflictRangeIndexMap;
+
+	Future<Void> releaseDelay;
+	Future<Void> releaseFuture;
+
+	std::vector<ResolveTransactionBatchReply> resolution;
+
+	double computeStart;
+	double computeDuration = 0;
+
+	Arena arena;
+
+	/// true if the batch is the 1st batch for this proxy, additional metadata
+	/// processing is involved for this batch.
+	bool isMyFirstBatch;
+	bool firstStateMutations;
+	
+	Optional<Value> oldCoordinators;
+
+	StoreCommit_t storeCommits;
+
+	std::vector<uint8_t> committed;
+
+	Optional<Key> lockedKey;
+	bool locked;
+
+	int commitCount = 0;
+
+	std::vector<int> nextTr;
+
+	bool lockedAfter;
+
+	Optional<Value> metadataVersionAfter;
+
+	int mutationCount = 0;
+	int mutationBytes = 0;
+
+	std::map<Key, MutationListRef> logRangeMutations;
+	Arena logRangeMutationsArena;
 
-	self->lastVersionTime = t1;
+	int transactionNum = 0;
+	int yieldBytes = 0;
 
-	++self->stats.commitBatchIn;
+	LogSystemDiskQueueAdapter::CommitMessage msg;
+
+	Future<Version> loggingComplete;
+
+	double commitStartTime;
+
+	Context(
+		ProxyCommitData*,
+		const std::vector<CommitTransactionRequest>*,
+		const int);
+
+	void setupDebugTrack();
+
+private:
+	void evaluateBatchSize();
+};
 
-	for (int t = 0; t<trs.size(); t++) {
-		if (trs[t].debugID.present()) {
-			if (!debugID.present())
+Context::Context(
+		ProxyCommitData* const pProxyCommitData_,
+		const std::vector<CommitTransactionRequest>* trs_,
+		const int currentBatchMemBytesCount) :
+
+			pProxyCommitData(pProxyCommitData_),
+			trs(std::move(*const_cast<std::vector<CommitTransactionRequest>*>(trs_))),
+			currentBatchMemBytesCount(currentBatchMemBytesCount),
+
+			startTime(g_network->now()),
+
+			localBatchNumber(++pProxyCommitData->localCommitBatchesStarted),
+			toCommit(pProxyCommitData->logSystem),
+			
+			committed(trs.size()) {
+
+	evaluateBatchSize();
+
+	if (batchOperations != 0) {
+		latencyBucket = std::min<int>(
+			SERVER_KNOBS->PROXY_COMPUTE_BUCKETS - 1,
+			SERVER_KNOBS->PROXY_COMPUTE_BUCKETS * batchBytes /
+				(batchOperations * (
+					CLIENT_KNOBS->VALUE_SIZE_LIMIT +
+					CLIENT_KNOBS->KEY_SIZE_LIMIT
+				))
+		);
+	}
+
+	// since we are using just the former to limit the number of versions actually in flight!
+	ASSERT(SERVER_KNOBS->MAX_READ_TRANSACTION_LIFE_VERSIONS <= SERVER_KNOBS->MAX_VERSIONS_IN_FLIGHT);
+}
+
+void Context::setupDebugTrack()  {
+	for (int t = 0; t < trs.size(); ++t) {
+        auto& tr = trs[t];
```

Comment:
Nit: you can use a more concise style
```suggestion
	for (const auto& tr : trs) {
```